src/main/java/org/nuxeo/wopi/jaxrs/WOPIRoot.java

```diff
@@ -45,6 +46,8 @@ public Object filesResource(@PathParam("fileId") FileInfo fileInfo) {
         WebContext context = getContext();
         context.setRepositoryName(fileInfo.repositoryName);
         CoreSession session = context.getCoreSession();
+        // flag the session's principal as a WOPI user for locking policy purpose
+        LockHelper.markAsWOPIUser(session.getPrincipal());
```

Comment:
This is going to be problematic as the principal actually comes from a cache in the authentication filter (actually the http session). This means this will leak beyond the duration of just this call. Also the Principal object is not supposed to be thread-safe... Let's discuss

Reply:
OK, let's suppose for now that the Office calls don't include a session cookie and that parallel requests won't be made for the same user.
For thread safety, let's synchronize the accesses to the context data map.